### PR TITLE
Raise a clearer error message when the user tries to comment on a draft post

### LINKED ISSUE #18386: :bug:  "Commentable is not valid on" on a Post in Draft status
# :footprints: Steps to reproduce

1. Create a post
2. Save it in draft status (ie. do not publish it)
3. Comment on it
4. Get the "Your comment could not be posted due to an error: Commentable is not valid. " message

image

# :pray: Expected behavior

Two options : 

:+1:  Ideally : be able to save the comment event on an unpublished post (which is quite useful in some cases for editor experience)
:-1:  Do not allow to comment... and then fix the bug

### PR DESCRIPTION
**Pull request by ChaelCodes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
The user experience when submitting an invalid comment on a draft PR is really confusing. They get the error message "Your comment could not be posted due to an error: Commentable is not valid." This PR updates the error message to "Your comment could not be posted due to an error: Commentable is not a published article"

## Related Tickets & Documents

- Related Issue #18386
- Closes #18386

## QA Instructions, Screenshots, Recordings

- Go to `/new`
- Fill in the title
- Click Save Draft
- Fill in the comment text area for that unpublished article
- Click submit on the comment
- [ ] Clearer Error message should be displayed

image

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

gif selected by bribery from jesus_h

### CODE REVIEW

**Comment by fdocr** [comment]
Thank you for this contribution @ChaelCodes and sorry for taking a minute to get back to you here. There are a few small rubocop errors in the spec file [annotated here in the PR](https://github.com/forem/forem/pull/18588/files) (you can also run `bundle exec rubocop spec/models/comment_spec.rb` to check and ensure they're fixed). 

Would you mind including these changes and also update the branch with `main`? In case it's helpful `git pull origin main` works just fine. We introduced some changes to CI and having the branch up to date with main would be great. If you don't have a chance to do this I believe I can push the changes to this branch next week and help get this merged soon. Thanks again!

**Comment by Ridhwana** [comment]
Closing and re-opening to trigger the CI-status-report

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -8,11 +8,13 @@
   include_examples "#sync_reactions_count", :article_comment
 
   describe "validations" do
-    subject { comment }
+    subject() { comment }
```
Comment (Ridhwana) [inline spec/models/comment_spec.rb:11]:
```suggestion
    subject { comment }
```

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -8,11 +8,13 @@
...
 
     describe "builtin validations" do
+      subject { build(:comment, user: user, commentable: article)}
```
Comment (Ridhwana) [inline spec/models/comment_spec.rb:14]:
```suggestion
      subject { build(:comment, user: user, commentable: article) }
```

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -42,6 +44,7 @@
         subject.commentable = build(:article, published: false)
 
         expect(subject).not_to be_valid
+        expect(subject.errors.full_messages).to include 'Commentable is not a published article'
```
Comment (Ridhwana) [inline spec/models/comment_spec.rb:47]:
```suggestion
        expect(subject.errors.full_messages).to include "Commentable is not a published article"
```

**Review by Ridhwana — COMMENTED** [review]
I've just added some suggestions to fix the Rubocop errors, @ChaelCodes feel free to accept them if they look good to you 😊

**Review by fdocr — APPROVED** [review]

**Pull request merged**